fetch.bs

```diff
@@ -6718,6 +6753,418 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+<var>activateAfter</var>, and <var>onActivatedWithoutTermination</var>, which is an algorithm that
+takes no arguments:
+
+<ol>
+ <li><p><a>Populate request from client</a> given <var>request</var>.
+
+ <li><p>Set <var>request</var>'s <a for=request>service-workers mode</a> to "<code>none</code>".
+
+ <li><p>Set <var>request</var>'s <a for=request>keepalive</a> to true.
+
+ <li><p>Let <var>deferredRecord</var> be a new <a>deferred fetch record</a> whose
+ <a for="deferred fetch record">request</a> is <var>request</var>.
+
+ <li>
+  <p><a for=list>Append</a> <var>deferredRecord</var> to <var>fetchGroup</var>'s
+  <a for="fetch group">deferred fetch records</a>.
+
+ <li>
+  <p>If <var>activateAfter</var> is non-null, then run the following steps <a>in parallel</a>:</p>
+
+  <ol>
+   <li>
+    <p>The user agent should wait until any of the following conditions is met:
+
+    <ul class=brief>
+     <li><p>At least <var>activateAfter</var> milliseconds have passed.
+
+     <li><p>The user agent has a reason to believe that it is about to lose the opportunity to
+     execute scripts, e.g., when the browser is moved to the background, or when
+     <var>request</var>'s <a for=request>client</a> is a {{Document}} that had a
+     "<code>hidden</code>" <a for=Document>visibility state</a> for a long period of time.
+    </ul>
+
+   <li><p>If the result of calling <a>process a deferred fetch</a> given <var>deferredRecord</var>
+   returns true, then <a>queue a global task</a> on the <a>deferred fetch task source</a> with
+   <var>request</var>'s <a for=request>client</a>'s
+   <a for="environment settings object">global object</a> to run
+   <var>onActivatedWithoutTermination</var>.
+  </ol>
+ </li>
+
+ <li><p>Return <var>deferredRecord</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To compute the <dfn>total request length</dfn> of a <a for=/>request</a> <var>request</var>:
+
+<ol>
+ <li><p>Let <var>totalRequestLength</var> be the <a for=string>length</a> of <var>request</var>'s
+ <a for=request>URL</a>, <a lt="URL serializer">serialized</a> with
+ [=URL serializer/exclude fragment=] set to true.
+
+ <li><p>Increment <var>totalRequestLength</var> by the  <a for=string>length</a> of
+ <var>request</var>'s <a for=request>referrer</a>, <a lt="URL serializer">serialized</a>.
+
+ <li><p><a for=list>For each</a> (<var>name</var>, <var>value</var>) of <var>request</var>'s
+ <a for=request>header list</a>, increment <var>totalRequestLength</var> by <var>name</var>'s
+ <a for="byte sequence">length</a> + <var>value</var>'s <a for="byte sequence">length</a>.
+
+ <li><p>Increment <var>totalRequestLength</var> by <var>request</var>'s <a for=request>body</a>'s
+ <a for=body>length</a>.
+
+ <li><p>Return <var>totalRequestLength</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn>process deferred fetches</dfn> given a <a>fetch group</a> <var>fetchGroup</var>:
+
+<ol>
+ <li><p><a for=list>For each</a> <a for="fetch group">deferred fetch record</a>
```

Comment:
> For each deferred fetch record deferredRecord of controlDocument’s fetch group’s deferred fetch records:

Consider the following example:


```
root (a.com) -> frame-1 (a.com) -> frame-2 (b.com)
             -> frame-3 (b.com) -> frame-4 (a.com)
```

Since `frame-1`'s `controlDocument` is `root`, any new fetchLater requests made in `frame-1`, when being processed by `To get the available deferred-fetch quota`, will only consider the existing deferred records from `root`? Is that expected?

Reply:
It's expected. Also when making the deferred fetch, the deferred fetch record would be added to root. frame-1 would never have its own deferred fetch records because it's not a control document.

Yes, in the example, the first one would succeed and the other two would fail.